include/meltpooldg/evaporation/evaporation_operation_marching_cube.hpp

```diff
@@ -0,0 +1,282 @@
+          ls.evaluate(true, true);
+
+          normal_vec.reinit(cell);
+          normal_vec.read_dof_values(normal_vector);
+          normal_vec.evaluate(true, false);
+
+          evap_flux.reinit(cell);
+          evap_flux.read_dof_values(evaporative_mass_flux);
+          evap_flux.evaluate(true, false);
+
+          for (unsigned int q_index = 0; q_index < ls.n_q_points; ++q_index)
+            {
+              const auto n_phi =
+                MeltPoolDG::VectorTools::normalize<dim>(normal_vec.get_value(q_index));
+
+              auto is_liquid = compare_and_apply_mask<SIMDComparison::less_than>(
+                ls.get_value(q_index),
+                VectorizedArray<double>(0.5),
+                evap_flux.get_value(q_index) / rho_g,
+                evap_flux.get_value(q_index) / rho_l);
+
+              evapor_vel[q_index] = is_liquid * n_phi;
+            }
+        }
+      level_set_as_heaviside.zero_out_ghosts();
+      normal_vector.zero_out_ghosts();
+      evaporative_mass_flux.zero_out_ghosts();
+
+      scratch_data.initialize_dof_vector(evaporation_velocity, evapor_vel_dof_idx);
+
+      /**
+       * write interface velocity to dof vector
+       */
+      UtilityFunctions::fill_dof_vector_from_cell_operation_vec<dim, dim>(
+        evaporation_velocity,
+        scratch_data.get_matrix_free(),
+        evapor_vel_dof_idx,
+        ls_quad_idx,
+        scratch_data.get_degree(evapor_vel_dof_idx),           // fe_degree of the resulting vector
+        scratch_data.get_degree(ls_hanging_nodes_dof_idx) + 1, // n_q_points_1d of cell operation
+        [&](const unsigned int cell,
+            const unsigned int quad) -> const Tensor<1, dim, VectorizedArray<double>> & {
+          return const_cast<const Tensor<1, dim, VectorizedArray<double>> &>(
+            evaporation_velocities[scratch_data.get_matrix_free().get_n_q_points(ls_quad_idx) *
+                                     cell +
+                                   quad]);
+        });
+
+      scratch_data.get_constraint(evapor_vel_dof_idx).distribute(evaporation_velocity);
+      scratch_data.get_pcout() << "    | evapor: |u|2 = " << evaporation_velocity.l2_norm()
+                               << std::endl;
+
+      evaporation_velocity.zero_out_ghosts();
+    }
+
+    static void
+    compute_mass_balance_source_term_sharp(const ScratchData<dim> &scratch_data,
+                                           VectorType &            mass_balance_rhs,
+                                           const VectorType &      evaporative_mass_flux,
+                                           const VectorType &      level_set_vector,
+                                           const double            rho_l,
+                                           const double            rho_g,
+                                           const unsigned int      evapor_dof_idx,
+                                           const unsigned int      pressure_dof_idx)
+    {
```

Comment:
I would handle the ifdef this way:
```suggestion
        (void)scratch_data;
        (void)mass_balance_rhs;
        (void)evaporative_mass_flux;
        (void)level_set_vector;
        (void)rho_l;
        (void)rho_g;
        (void)evapor_dof_idx;
        (void)pressure_dof_idx;
        #ifdef MELT_POOL_DG_WITH_ADAFLO
```